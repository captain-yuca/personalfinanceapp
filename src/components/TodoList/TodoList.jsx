import React from 'react'
import PropTypes from 'prop-types'

import TodoItem from '../TodoItem'

import { Wrapper } from './TodoList.styles'

const propTypes = {
  todos: PropTypes.arrayOf(PropTypes.object),
  removeTodo: PropTypes.func,
}

function TodoList({ todos, removeTodo }) {
  if (!todos.length) return <div className="empty">No todos :)</div>

  return (
    <Wrapper>
      {todos.map(todo => (
        <TodoItem key={todo.id} todo={todo} removeTodo={removeTodo} />
      ))}
    </Wrapper>
  )
}

TodoList.propTypes = propTypes

export default TodoList
