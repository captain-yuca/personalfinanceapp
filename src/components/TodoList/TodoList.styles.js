import styled from 'styled-components'

export const Wrapper = styled.div`
  margin-top: 24px;
  border: 2px solid ${({ theme }) => theme.colors.primary};
  border-radius: 8px;
  width: 300px;
`
