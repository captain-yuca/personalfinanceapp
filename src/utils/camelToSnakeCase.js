export const camelToSnakeCase = str => str.replace(/([A-Z])/g, group => `_${group.toLowerCase()}`)
