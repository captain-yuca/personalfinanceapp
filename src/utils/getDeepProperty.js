export const getDeepProperty = (path, obj) =>
  path.split('.').reduce((prev, curr) => (prev ? prev[curr] : null), obj || null)
