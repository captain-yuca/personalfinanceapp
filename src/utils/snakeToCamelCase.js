export const snakeToCamelCase = str =>
  str.replace(/([-_][a-z])/g, group => group.toUpperCase().replace('-', '').replace('_', ''))
