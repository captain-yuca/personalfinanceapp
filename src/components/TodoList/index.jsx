import React from 'react'
import TodoList from './TodoList'

function Wrapper(props) {
  return <TodoList {...props} />
}

export default Wrapper
