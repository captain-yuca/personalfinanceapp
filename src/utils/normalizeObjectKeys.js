import { camelToSnakeCase } from './camelToSnakeCase'
import { snakeToCamelCase } from './snakeToCamelCase'

export const normalizeObjectKeys = (obj, toSnake = false) =>
  Object.entries(obj).reduce((accum, [rawKey, value]) => {
    const key = toSnake ? camelToSnakeCase(rawKey) : snakeToCamelCase(rawKey)
    return { ...accum, [key]: value }
  }, {})
