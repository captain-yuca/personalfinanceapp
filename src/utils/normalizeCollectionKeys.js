import { normalizeObjectKeys } from './normalizeObjectKeys'

export const normalizeCollectionKeys = (collection, toSnake = false) =>
  collection.map(item => normalizeObjectKeys(item, toSnake))
