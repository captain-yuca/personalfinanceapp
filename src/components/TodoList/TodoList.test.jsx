import React from 'react'
import { render } from '../../test.utils'

import TodoList from './TodoList'

describe('TodoList component', () => {
  const props = {
    todos: [
      {
        id: 1,
        text: 'Test todo',
      },
    ],
    removeTodo: jest.fn(),
  }

  it('should render correctly', () => {
    const { container, getByText } = render(<TodoList {...props} />)

    expect(getByText('Test todo')).toBeInTheDocument()
    expect(container).toMatchSnapshot()
  })
})
