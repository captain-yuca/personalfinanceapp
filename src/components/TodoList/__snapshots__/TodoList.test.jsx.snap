// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoList component should render correctly 1`] = `
.c3 {
  color: #ffffff;
  font-size: 14px;
  -webkit-transition: all 0.3s ease;
  transition: all 0.3s ease;
}

.c5 {
  color: #ffffff;
  font-weight: bold;
  -webkit-transition: all 0.3s ease;
  transition: all 0.3s ease;
}

.c1 {
  border-bottom: 2px solid #ff0000;
  padding: 8px 16px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  cursor: pointer;
}

.c1:hover .c2,
.c1:hover .c4 {
  color: #ff0000;
}

.c0 {
  margin-top: 24px;
  border: 2px solid #ff0000;
  border-radius: 8px;
  width: 300px;
}

<div>
  <div
    class="c0"
  >
    <div
      class="c1"
    >
      <span
        class="c2 c3"
      >
        Test todo
      </span>
      <span
        class="c4 c5"
      >
        X
      </span>
    </div>
  </div>
</div>
`;
